### PR TITLE
Handle audit log path in /var/log/origin

### PR DESCRIPTION
**Pull request by rsevilla87**
This commit is intended to make the installer able to handle OpenShift audit log paths different from (/etc/origin/master, /var/lib/origin, /etc/origin/cloudprovider), which in most of cases are not appropriate since those directories are not usually used as log store.

The code included create a new directory /var/log/origin if auditFilePath is defined within this directory, and adds it as a hostPath in the apiserver static pod definition. custom module master_check_paths_in_config has been also properly tuned to allow this directories to be defined in the master-config.yaml

If accepted, this change should be properly documented.

More info: https://github.com/openshift/openshift-ansible/pull/10715

### CODE REVIEW

**Comment by sdodson** [comment]
@openshift/sig-master @smarterclayton Do we want to add this? It seems like a reasonable solution for what to do about audit logs in static pods but it likely doesn't address the problem in all possible components that may be respecting the config options for auditlogging. Like service catalog.

File: roles/lib_utils/action_plugins/master_check_paths_in_config.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
 ITEMS_TO_POP = (
     ('oauthConfig', 'identityProviders'),
+    ('auditConfig', 'auditFilePath'),
```
Comment (michaelgugino) [inline roles/lib_utils/action_plugins/master_check_paths_in_config.py]: This is not the appopriate place for this.  We should utilized ALLOWED_DIRS instead.

**Comment by juanluisvaladas** [comment]
> I'm not a fan of this change. It would have been a good idea when we set out for the static pods, but if someone with an existing cluster scales up, they'll now have masters that are configured differently.

Isn't this statement true for every option?

**Comment by soltysh** [comment]
> @openshift/sig-master @smarterclayton Do we want to add this? It seems like a reasonable solution for what to do about audit logs in static pods but it likely doesn't address the problem in all possible components that may be respecting the config options for auditlogging. Like service catalog.

From openshift API pov this makes sense, I'm not 100% sure about other components, though.

**Comment by gregswift** [comment]
Please backport this to 3.10 since its part of the upgrade path to 3.11

**Comment by sdodson** [comment]
/lgtm
/cherrypick release-3.10 
Looks safe enough, only triggered if they actually set audit file path to /var/log/origin.

**Comment by juanluisvaladas** [comment]
02267601

**Pull request closed without merging**